torch/distributed/_tensor/sharding_prop.py

```diff
@@ -276,11 +276,12 @@ def propagate(self, op_overload: OpOverload, op_schema: OpSchema) -> OutputShard
         Propagate the sharding for an operator given the op_schema.
         Cache the propagation results to avoid running propagation again.
         """
-        if op_schema in self.cached_prop_results:
-            return self.cached_prop_results[op_schema]
+        op_schema_key = str(op_schema)
```

Comment:
yes agree. Upon checking the trace, str is also expensive. We want to finally get rid of this str but somehow need to ensure `op_schema` does not get changed.

Reply:
op_schema get changed because of `tensor_meta` got changed? if we change the `__eq__` to only verify tensor_meta shape/stride, would it still be an issue?

Reply:
No, not `tensor_meta` what I have seen is the placements also get changed.